src/transformers/models/bart/modeling_bart.py

```diff
@@ -263,14 +287,229 @@ def forward(
         return attn_output, attn_weights_reshaped, past_key_value
 
 
+class BartFlashAttention2(BartAttention):
+    """
+    Bart flash attention module. This module inherits from `BartAttention` as the weights of the module stays
+    untouched. The only required change would be on the forward pass where it needs to correctly call the public API of
+    flash attention and deal with padding tokens in case the input contains any of them.
+    """
+
+    def _shape(self, tensor: torch.Tensor, seq_len: int, bsz: int):
+        return tensor.view(bsz, seq_len, self.num_heads, self.head_dim)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        key_value_states: Optional[torch.Tensor] = None,
+        past_key_value: Optional[Tuple[torch.Tensor]] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        layer_head_mask: Optional[torch.Tensor] = None,
+        output_attentions: bool = False,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
+        # BartFlashAttention2 attention does not support output_attentions
+        output_attentions = False
```

Comment:
I think it would be better to raise an exception here, rather than silently override

```suggestion
        if output_attentions:
            raise ValueError("BartFlashAttention2 attention does not support output_attentions")
```

Reply:
Agree, it would become a bit out of sync with Llama though, but ok! 
https://github.com/huggingface/transformers/blob/c9e72f55b2dc4b9be4edb986dce0552582b328f2/src/transformers/models/llama/modeling_llama.py#L451
We should probably deprecate the "silent ignore" in Llama at some point!